pkg/internal/httpclient/helper.go

```diff
@@ -0,0 +1,160 @@
+package httpclient
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/mercadopago/sdk-go/pkg/config"
+)
+
+const (
+	currentSDKVersion string = "x.x.x"
+	productID         string = "abc"
+)
+
+var (
+	pathParamFormat = regexp.MustCompile("[{}]")
+
+	userAgent  = fmt.Sprintf("MercadoPago Go SDK/%s", currentSDKVersion)
+	trackingID = fmt.Sprintf("platform:%s,type:SDK%s,so;", runtime.Version(), currentSDKVersion)
+)
+
+type CallData struct {
+	Body        any
+	PathParams  map[string]string
+	QueryParams map[string]string
+
+	Method string
+	URL    string
+}
+
+func Run[T any](ctx context.Context, cfg *config.Config, callData CallData) (T, error) {
+	var result T
+
+	req, err := createRequest(ctx, cfg, callData)
+	if err != nil {
+		return result, err
+	}
+
+	b, err := Send(cfg.Requester, req)
+	if err != nil {
+		return result, err
+	}
+
+	return wrapUnmarshal(b, result)
```

Comment:
Tudo bem que é um wrap, mas acredito que podemos deixar o nome mais simples
```suggestion
	return unmarshal(b, result)
```

Reply:
feito!